src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -882,13 +883,47 @@ void Graph::AllocateWithReuse() {
             }
         }
         for (auto& group : groups) {
-            auto grpMemMngr =
+            MemoryMngrPtr grpMemMngr;
+            grpMemMngr =
                 std::make_shared<DnnlMemoryMngr>(make_unique<MemoryMngrWithReuse>());
+            // deternmine a group with outputs.
+            size_t isOutGrp = 0;
+            int64_t outBoxId = -1;
+            for (auto& box : group) {
+                if (std::any_of(
+                    edge_clusters[box.id].begin(),
+                    edge_clusters[box.id].end(),
+                    [box](const ov::intel_cpu::EdgePtr edge) {
+                        return edge->getChild()->getType() == Type::Output;
+                    })) {
+                        isOutGrp++;
+                        outBoxId = box.id;
+                }
+            }
+            if (isOutGrp) {
+                IE_ASSERT(isOutGrp==1);  // reuse_io_tensors false
+                grpMemMngr =
+                    std::make_shared<OutputMemoryMngr>(grpMemMngr);
+                DEBUG_LOG(grpMemMngr, " ", this);
+
+                // Store the output memory managers.
+                // So that, the infer requests can be able to access them.
+                for (auto& edge : edge_clusters[outBoxId]) {
+                    const auto child = edge->getChild();
+                    if (child->getType() == Type::Output) {
+                        for (auto &output : outputNodesMap) {
+                            if (output.second == child) outputNodesMemMngrMap[output.first] = grpMemMngr;
+                        }
+                    }
+                }
+            }
```

Comment:
@maxnick indeed not a good place to add OutputMemoryMngr. I just realize that there might be situations when graph input is dynamic shape, but output is static shape (e.g. some interpolation ops). 
Need further discussion to follow up https://github.com/maxnick/openvino/pull/62/#discussion_r1228169716

Reply:
Also there could be corner cases that more than one outputs share the same memory. e.g. each output is the output from Split.

Reply:
> Also there could be corner cases that more than one outputs share the same memory. e.g. each output is the output from Split.

That is why I recommended check `Edge::Status::NeedAllocation` status. If the edge status is different, it means it shares the memory with other edges and in that case we cann't set proxy memory manager.

>I just realize that there might be situations when graph input is dynamic shape, but output is static shape (e.g. some interpolation ops).

To avoid affecting static tensors, this step may be performed after initializing static tensors, L831.

Reply:
done